src/components/RoomDropdown/index.tsx

```diff
@@ -0,0 +1,356 @@
+import React, { useEffect, useRef } from 'react'
+import { useState } from 'react'
+import ReactModal from 'react-modal'
+import styled from 'styled-components'
+import { KARELIA } from '../../utils/css'
+import MemberModal from '../MemberModal'
+import MemberOptions from '../MemberModal/MemberOptions'
+
+type Props = {
+    button?: any
+}
+
+const customStyles = {
+    overlay: {
+        backgroundColor: 'rgba(0, 0, 0, 0.3)',
+        zIndex: '1000',
+    },
+    content: {
+        border: 'none',
+        width: '528px',
+        height: '456px',
+        top: '50%',
+        left: '50%',
+        right: 'auto',
+        bottom: 'auto',
+        borderRadius: '20px',
+        transform: 'translate(-50%, -50%)',
+        fontFamily: `${KARELIA}`,
+    },
+}
+
+const members = [
+    {
+        address: '0x71C7656EC7ab88b098defB751B7401B5f6d8976F',
+    },
+    {
+        address: '0x71C7656EC7ab88b098defB751B7401B5f6d8976F',
+    },
+    {
+        address: '0x71C7656EC7ab88b098defB751B7401B5f6d8976F',
+    },
+    {
+        address: '0x71C7656EC7ab88b098defB751B7401B5f6d8976F',
+    },
+    {
+        address: '0x71C7656EC7ab88b098defB751B7401B5f6d8976F',
+    },
+]
+
+const DropDownContainer = styled.div`
+    margin-left: 10px;
+`
+const DropDownHeader = styled.div`
+    margin-bottom: 0.8em;
+    padding: 0 2em 0 1em;
+    box-shadow: 0 2px 3px rgba(0, 0, 0, 0.15);
+    font-weight: 500;
+    font-size: 1.3rem;
+    color: #3faffa;
+    background: #ffffff;
+`
+const DropDownListContainer = styled.div`
+    position: absolute;
+    top: 75px;
+    right: 35px;
+    z-index: 100;
+`
+const DropDownList = styled.table`
+    width: 250px;
+    padding: 0;
+    margin: 0;
+    background: #ffffff;
+    border-radius: 10px;
+    box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.1);
+    box-sizing: border-box;
+    color: #59799c;
+    font-family: ${KARELIA};
+    font-size: 14px;
+    list-style-type: none;
+    border-spacing: 0px;
+    hr {
+        border: 0;
+        border-top: 1px solid #dee6ee;
+        padding: 0px;
+        margin: 0px;
+    }
+`
+const ListItem = styled.tr`
+    padding: 10px 15px;
+    background-clip: padding-box;
+    display: flex;
+    td {
+        line-height: 13px;
+        &:last-child {
+            margin-left: 15px;
+        }
+    }
+    &:first-child {
+        padding-top: 15px;
+        border-radius: 10px 10px 0 0;
+    }
+    &:last-child {
+        padding-bottom: 15px;
+        border-radius: 0 0 10px 10px;
+    }
+    :hover {
+        background-color: #f1f4f7;
+    }
+`
+
+const StyledModalContent = styled.div`
+    .contentContainer {
```

Comment:
Those are real class names used in our code (in `RoomDropdown`). Let's turn them into styled components, too.

Reply:
Not that familiar with styled-components best practices but should I avoid using classnames and instead make all of the child components styled-components?

Reply:
Yes*, please, for consistency. It'll also just appreciate in value over time (no need to keep an eye out for name conflicts and so on).

*) In practice not all child components have to be converted. Depends on the situation. Example

```tsx
const MyDivo = styled.div`
    color: red;

    h1 {
        color: blue;
    }
`

const div = (
    <MyDivo>
        <h1>Hello world</h1>
    </MyDivo>
)
```
If there's just one `h1` then there's no need for a separate class. If you can come up with a simple but deterministic selector for an element then there's no need for a class.

---

You may already know this… You can style styled-components within other styled-components. Example

```tsx
const A = styled.div``

const B = styled.div`
    ${A} {
        color: red;
    }
`
```

It's neat.

Reply:
This is really helpful, thanks for the tips!